### PR TITLE
Add unique naming to hardcoded resources to fix pipeline crashing

### PR DESCRIPTION
**Pull request by patilsnr**
## Purpose

* Adds GUID to resources created in samples to prevent hardcoded ghost configs from remaining in testing hub and crashing testing pipeline.
* Misc IDE suggestions

### CODE REVIEW

File: iot-hub/Samples/service/AutomaticDeviceManagementSample/AutomaticDeviceManagementSample.cs [hunk 1]
```diff
@@ -25,44 +25,47 @@ public AutomaticDeviceManagementSample(RegistryManager registryManager)
...
-            await AddDeviceConfiguration("config002").ConfigureAwait(false);
-            await AddDeviceConfiguration("config003").ConfigureAwait(false);
-            await AddDeviceConfiguration("config004").ConfigureAwait(false);
-            await AddDeviceConfiguration("config005").ConfigureAwait(false);
+
+            // save unique config names to be used for deletion
+            var configs = new List<string>(5);
+            for (int i = 0; i < 5; i++)
+            {
+                configs.Add($"config00{i+1}_{Guid.NewGuid()}");
```
Comment (brycewang-microsoft) [inline iot-hub/Samples/service/AutomaticDeviceManagementSample/AutomaticDeviceManagementSample.cs:33]:
```suggestion
                configs.Add($"config00{i}_{Guid.NewGuid()}");

```
How about we index starting from 000?

**Review by brycewang-microsoft — APPROVED** [review]

File: iot-hub/Samples/service/AutomaticDeviceManagementSample/AutomaticDeviceManagementSample.cs [hunk 2]
```diff
@@ -25,44 +25,53 @@ public AutomaticDeviceManagementSample(RegistryManager registryManager)
...
-            await AddDeviceConfiguration("config001").ConfigureAwait(false);
-            await AddDeviceConfiguration("config002").ConfigureAwait(false);
-            await AddDeviceConfiguration("config003").ConfigureAwait(false);
-            await AddDeviceConfiguration("config004").ConfigureAwait(false);
-            await AddDeviceConfiguration("config005").ConfigureAwait(false);
 
-            Console.WriteLine("List existing configurations");
-            await GetConfigurations(5).ConfigureAwait(false);
+            // save unique config names to be used for deletion
+            var configs = new List<string>(5);
```
Comment (drwill-ms) [inline iot-hub/Samples/service/AutomaticDeviceManagementSample/AutomaticDeviceManagementSample.cs:30]: Use a constant:

```csharp
const int configsToAdd = 5;
var configs = new List<string>(configsToAdd);

for (int i = 0; i < configsToAdd; ++i)
```

File: iot-hub/Samples/service/AutomaticDeviceManagementSample/AutomaticDeviceManagementSample.cs (hunk 1 above)
Comment (patilsnr) [reply iot-hub/Samples/service/AutomaticDeviceManagementSample/AutomaticDeviceManagementSample.cs]: Fine with me, this was just making it as close as possible to the original naming scheme. I agree indexing from zero is cleaner.

File: iot-hub/Samples/service/AutomaticDeviceManagementSample/AutomaticDeviceManagementSample.cs [hunk 3]
```diff
@@ -25,44 +25,53 @@ public AutomaticDeviceManagementSample(RegistryManager registryManager)
...
 
-            Console.WriteLine("Remove some connfigurations");
-            await DeleteConfiguration("config004").ConfigureAwait(false);
-            await DeleteConfiguration("config002").ConfigureAwait(false);
+            try
+            {
+                for (int i = 0; i < 5; i++)
+                {
+                    configs.Add($"config00{i + 1}_{Guid.NewGuid()}");
```
Comment (drwill-ms) [inline iot-hub/Samples/service/AutomaticDeviceManagementSample/AutomaticDeviceManagementSample.cs:36]: What's wrong with a config named 000? :p

**Review by drwill-ms — APPROVED** [review]
Does this sample need to demonstrate with 5 configs? How many are necessary?

File: iot-hub/Samples/service/AutomaticDeviceManagementSample/AutomaticDeviceManagementSample.cs [hunk 4]
```diff
@@ -25,44 +25,54 @@ public AutomaticDeviceManagementSample(RegistryManager registryManager)
...
+            var configs = new List<string>(configsToAdd);
 
-            Console.WriteLine("Remove some connfigurations");
-            await DeleteConfiguration("config004").ConfigureAwait(false);
-            await DeleteConfiguration("config002").ConfigureAwait(false);
+            try
+            {
+                for (int i = 0; i < 5; i++)
+                {
+                    configs.Add($"config00{i}_{Guid.NewGuid()}");
```
Comment (andyk-ms) [inline iot-hub/Samples/service/AutomaticDeviceManagementSample/AutomaticDeviceManagementSample.cs:37]: Probably don't need "00" but no issue.

**Review by andyk-ms — APPROVED** [review]

**Pull request merged**